doc/new_boilerplate_migration.md

```diff
@@ -0,0 +1,635 @@
+  },
+  $fooPropsConfig, // ignore: undefined_identifier
+); 
+
+mixin FooProps on UiProps {
+  String foo;
+}
+```
+
+Here, `uiFunctionComponent` gets a generic parameter of `FooProps` inferred 
+from the LHS typing, allowing props to be statically typed as `FooProps`.
+
+The generated `$FooPropsConfig` is passed in as an argument, and serves 
+as the entrypoint to the generated code. 
+
+#### With Default Props
+
+`defaultProps` on function components is 
+[already deprecated](https://github.com/facebook/react/pull/16210).
+
+Use null-aware operators to default null values. This provides almost the
+same behavior as `defaultProps`, but with the restriction that a given prop
+__must either be nullable or have a default value, but not both__.
+
+```dart
+UiFactory<FooProps> Foo = uiFunctionComponent(
+  (props) {
+    final foo = props.foo ?? 'default foo value';
+
+    return 'foo: $foo'; 
+  },
+  $fooPropsConfig, // ignore: undefined_identifier
+); 
+```
+
+#### With propTypes
+
+```dart
+UiFactory<FooProps> Foo = uiFunctionComponent(
+  (props) {
+    return 'foo: ${props.foo}'; 
+  }, 
+  $fooPropsConfig, // ignore: undefined_identifier
+  getPropTypes: (keyFor) => {
+    keyFor((p) => p.foo): (props, info) {
+      if (props.foo == 'bar') {
+        return PropError('You can\'t foo a bar, silly');
+      }
+    }
+  },
+);
+```
+
+`getPropTypes` provides a way to set up prop validation within the 
+same variable initializer.
+
+#### Local function components using just a props mixin (no top-level Factory necessary)
+
+```dart
+import 'package:over_react/over_react.dart';
+
+part 'foo.over_react.g.dart';
+
+mixin FooProps on UiProps {
+  String foo;
+}
+
+// Example function; this can look like anything and doesn't have to 
+// be declared in this file. 
+UiFactory<FooProps> createFooHoc(UiFactory otherFactory) { 
+  Object closureVariable; 
+  // ... 
+
+  final FooHoc = uiFunctionComponent<FooProps>(
+    (props) { 
+      return otherFactory()( 
+        Dom.div()('closureVariable: ${closureVariable}'), 
+        Dom.div()('prop foo: ${props.foo}'), 
+      );   
+    },
+    $fooPropsConfig, // ignore: undefined_identifier 
+  ); 
+
+  return FooHoc; 
+} 
+```
+
+## Upgrading
+
+To update your repository to the new boilerplate, you can use 
+[over_react_codemod](https://github.com/Workiva/over_react_codemod)'s 
+`boilerplate_upgrade` executable to make it easier. This codemod goes 
+through the repository and updates the boilerplate as necessary. While 
+the codemod will handle many basic updates, it will still need to be 
+supplemented with some manual checks and refactoring. 
+
+Before running the codemod, run `semver_audit` inside your repository 
+and save the report using the following commands:
+
+1. `pub global activate semver_audit --hosted-url=https://pub.workiva.org`
+2. `pub global run semver_audit generate 2> semver_report.json`
+
+This will allow the codemod to check whether or not components are
+public API.
```

Comment:
```suggestion
If you are migrating a Workiva library, before running the codemod, run `semver_audit` inside your repository 
and save the report using the following commands:

1. `pub global activate semver_audit --hosted-url=https://pub.workiva.org`
2. `pub global run semver_audit generate 2> semver_report.json`

This will allow the codemod to check whether or not components are
public API.

If you are migrating a library outside of the Workiva ecosystem, run the command 
below with the `--treat-all-components-as-private` flag.
```